.github/actions/dockerScoutScan/action.yml

```
@@ -0,0 +1,57 @@
+name: Docker scout scan
+description: Comparison of Docker images using Docker Scout
+
+inputs:
+  username:
+    description: Username for logging into the Docker Scout
+    required: true
+  token:
+    description: PAT token for logging into the Docker Scout
+    required: true
+  github-token:
+    description: GitHub token for commenting the PR
+    required: true
+  current-image:
+    description: Image to be compared
+    required: true
+  current-tag:
+    description: Tag of the image to be compared
+    required: true
+  compare-to-image:
+    description: Image name to compare to
+    required: true
+  compare-to-tag:
+    description: Image to compare to, required if `compare-to-regex` is not provided, if both provided, this is prioritized
+    required: false
+  compare-to-regex:
+    description: Regex to find latest image in the registry to compare to, required if `compare-to` is not provided
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Ensure `compare-to-tag` or `compare-to-regex` is provided
+      if: ${{ !inputs.compare-to-tag && !inputs.compare-to-regex }}
+      shell: bash
+      run: exit 1
+
+    - name: Log into registry with PAT to use SCOUT.
+      uses: docker/login-action@v3
+      with:
+        username: ${{ inputs.username }}
+        password: ${{ inputs.token }}
+
+    - name: Get latest tag
```

```suggestion
    - name: Get tag by regex
```
Maybe? As there is no explicit condition on that tag being also the latest.

---

ultranit: I'd also consider naming the input `compare-to-regex-tag` or similar. With just 'regex' i get the notion the full name will be checked by regex (as the competing input is named `compare-to-tag`)..

---

another thought, bound to happen in potential follow-up. We cannot use version.go as for deployment it will have updated and unreleased version within, but at this point we should be able to say what was the latest successful release (from GH release/tag).. each step could use that info and just append the specific distribution string. This way we could get rid of querying all the tags (repeatedly, for each step utilizing this action, I presume) and manual filtering. 

(Of course this hypothesis works with our belief in the deployment working as it should.. but I presume that is where we want to head anyway).